iml-api/src/graphql.rs

```diff
@@ -431,13 +441,45 @@ impl QueryRoot {
             "#
         )
         .fetch(&context.pg_pool)
+        .map_ok(|x| SnapshotRetention {
+            id: x.id,
+            filesystem_name: x.filesystem_name,
+            delete_when: DeleteWhen {
+                value: x.delete_num,
+                unit: x.delete_unit,
+            },
+            last_run: x.last_run,
+            keep_num: x.keep_num,
+        })
         .try_collect()
         .await?;
 
         Ok(xs)
     }
 }
 
+struct SnapshotIntervalName {
+    id: i32,
+    fs_name: String,
+    timestamp: DateTime<Utc>,
+}
+
+fn parse_snapshot_name(name: &str) -> Option<SnapshotIntervalName> {
+    let name_parts: Vec<&str> = name.split("-").collect();
```

Comment:
nit, could probably pattern match the slice which would probably read cleaner:

https://doc.rust-lang.org/edition-guide/rust-2018/slice-patterns.html